File: xiaomi/project/repository-of-wangke/MyApplication/app/src/main/java/com/example/myapplication/work0602/MainActivity.java
package com.example.myapplication.work0602;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.myapplication.R;
import com.example.myapplication.work0603.SearchActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.work0602_activity_main);
        TextView gotoPhone = findViewById(R.id.work0602_goto_phone);
        gotoPhone.setOnClickListener(this);
        TextView gotoSearch = findViewById(R.id.work0602_goto_search);
        gotoSearch.setOnClickListener(this);

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void onClick(View v) {
        Intent intent;
        switch (v.getId()) {
            case R.id.work0602_goto_phone:
                intent = new Intent(MainActivity.this, PhonePermissionActivity.class);
                startActivity(intent);
                break;
            case R.id.work0602_goto_search:
                intent = new Intent(MainActivity.this, SearchActivity.class);
                startActivity(intent);
                break;
        }
    }
}

File: AndroidDemo/CoroutinesDemo/app/src/main/java/com/example/coroutinesdemo/MainActivity.kt
package com.example.coroutinesdemo

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val button1:Button = findViewById(R.id.button1)
        button1.setOnClickListener {
            val data = "test string"
            val intent = Intent(this,SecondActivity::class.java)
            intent.putExtra("extra_data",data)
            startActivity(intent)
        }

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu,menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when(item.itemId){
            R.id.add_item-> Toast.makeText(this,"Add clicked",Toast.LENGTH_SHORT).show()
            R.id.remove_item-> Toast.makeText(this,"Remove clicked",Toast.LENGTH_SHORT).show()
        }
        return true
    }
}